src/main/java/vectorwing/farmersdelight/common/block/RopeBlock.java

```diff
@@ -46,21 +49,59 @@ public BlockState getStateForPlacement(BlockPlaceContext context) {
 		return state != null ? state.setValue(TIED_TO_BELL, world.getBlockState(posAbove).getBlock() == Blocks.BELL) : null;
 	}
 
+	public int findSlotForRope(Inventory inventory) {
```

Comment:
The reason why I didn't use `Inventory#add(itemStack)` is that this method would always place the stack in player's **first empty slot** if there's no stack to combine. In that case, it would cause slightly misfunction as the first empty slot would most likely to be the **main hand**. Placing stack in main hand would block the reeling action, and the player would have to stop and change main hand to another slot.
Considering that the chance of a player to reel the rope without holding any is actually quite small, I would like to change my original idea and follow your suggestion :)

Reply:
That's actually a fair point, didn't think of that. It would most likely prioritize the hotbar, speeding up the blockage of an empty hand, so actually looking into the inventory first makes sense.

I'll decide later which one to go for, but thanks for updating either way. The method could probably reside in ItemUtils as well, given it may be useful for other cases.